cpp/src/arrow/type.cc

```diff
@@ -219,11 +219,13 @@ std::string LargeListType::ToString() const {
 
 MapType::MapType(const std::shared_ptr<DataType>& key_type,
                  const std::shared_ptr<DataType>& item_type, bool keys_sorted)
+    : MapType(key_type, std::make_shared<Field>("value", item_type), keys_sorted) {}
```

Comment:
Instead of `std::make_shared<Field>`, we can use `field("value", item_type)` directly.